parsons/sisense/sisense.py

```diff
@@ -0,0 +1,106 @@
+        This method generates a dashboard URL following the approach described in the
+        `Sisense Support guide <https://support.sisense.com/hc/en-us/community/posts/360038329993-Embed-URL-Generation-Made-Easy-With-Python-Script>`_ # noqa
+        This allows the Sisense API to create the URL from a JSON blob,
+        which is less error prone than creating the URL manually.
+
+        `Args:`
+            payload: dict
+                A JSON blob defining the dashboard to publish.
+        `Returns:`
+            A URL or an error.
+        """
+        response = self.api.post_request('shared_dashboard/create', data=json.dumps(payload))
+        try:
+            return response['url']
+        except KeyError as error:
+            logger.error('Failed to generate embedded dashboard URL.')
+            raise error
+
+    def get_dashboards(self, dashboard_id=None, params=None):
+        """
+        `Args:`
+            dashboard_id: int or str
+                The dashboard ID. If no dashboard ID is provided, all dashboards
+                for the user are returned.
+            params: dict
+                Request parameters. Check the `Sisense API documentation <https://sisense.dev/reference/rest/v1.html>`_ for options. # noqa
+        `Returns:`
+            A Table or an error.
+        """
+        if dashboard_id is None:
+            response = self.api.get_request('dashboards', params=params)
+        else:
+            response = list(self.api.get_request(f'dashboards/{dashboard_id}', params=params))
+        return Table(response)
+
+    def get_dashboard_shares(self, dashboard_id, share_id=None, params=None):
+        """
+        `Args:`
+            dashboard_id: int or str (required)
+                The dashboard ID.
+            share_id: int or str
+                The share ID. If no share ID is provided, all shares
+                for the dashboard are returned.
+            params: dict
+                Request parameters. Check the `Sisense API documentation <https://sisense.dev/reference/rest/v1.html>`_ for options. # noqa
+        `Returns:`
+            A Table or an error.
+        """
+        if share_id is None:
+            response = self.api.get_request(f'dashboards/{dashboard_id}/shares', params=params)
+        else:
+            response = list(self.api.get_request(f'dashboards/{dashboard_id}/shares/{share_id}',
+                                                 params=params))
+        return Table(response)
+
+    def publish_dashboard(self, payload=None):
+        """
+        This method publishes a dashboard based on a URL encoded JSON blob. For available options, see the `API documentation <https://dtdocs.sisense.com/article/embed-api-options>`_. # noqa
+
+        `Args:`
+            payload: dict
+                A JSON blob defining the dashboard to publish.
+        `Returns:`
+            Response or an error
+        """
+        return requests.post(self._create_dashboard_url(payload))
```

Comment:
I don't think you need this `post`.. the `_create_dashboard_url` method does the work of creating the dashboard, and returns the URL that I think the user will want.